components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -255,6 +260,21 @@ bool AIChatTabHelper::HasPageContent() {
   return !article_text_.empty();
 }
 
+void AIChatTabHelper::DisconnectPageContents() {
+  CleanUp();
+
+  should_page_content_be_disconnected_ = true;
+}
+
+void AIChatTabHelper::ClearConversationHistory() {
+  chat_history_.clear();
```

Comment:
Shouldn't we call `ClearAllQueries` here too? I tested submitting a human prompt, then hitting clear before the response was received and the UI showed the response to a cleared prompt, which was a weird state to see.

Reply:
Added